File: da-test/src/test/java/ro/johann/da/test/api/service/decision/transfer/PropertyOutput.java
package ro.johann.da.test.api.service.decision.transfer;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.experimental.FieldDefaults;

import static lombok.AccessLevel.PRIVATE;

@Data
@FieldDefaults(level = PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class PropertyOutput {
  String alternativeId;
  String criteriaId;
  Float value;
  Float utility;
  Integer rank;
}

File: da-decision-service/src/main/kotlin/ro/johann/da/decision/api/transfer/PropertyOutput.kt
package ro.johann.da.decision.api.transfer

import com.fasterxml.jackson.annotation.JsonFormat
import ro.johann.da.decision.domain.ProcessedEntity
import ro.johann.da.decision.domain.Property
import java.time.LocalDateTime
import java.util.UUID

data class PropertyOutput(
  val alternativeId: UUID,
  val criteriaId: UUID,
  val value: Double,
  val rank: Int?,
  val utility: Double?,
  @JsonFormat(pattern = "dd-MM-yyyy hh:mm:ss")
  val createdAt: LocalDateTime,
  @JsonFormat(pattern = "dd-MM-yyyy hh:mm:ss")
  val updatedAt: LocalDateTime
) {
  constructor(property: Property, processedEntity: ProcessedEntity? = null) : this(
    alternativeId = property.alternative.id,
    criteriaId = property.criteria.id,
    value = property.value,
    rank = processedEntity?.rank,
    utility = processedEntity?.utility,
    createdAt = property.createdAt,
    updatedAt = property.updatedAt
  )
}
